GeoLib/Polyline.h

```diff
@@ -50,6 +51,109 @@ enum class Location
 class Polyline : public GeoObject
 {
 public:
+	class SegmentIterator
+	    : public std::iterator<std::forward_iterator_tag, LineSegment>
```

Comment:
Could you put the iterator class in a seperate file? This takes up almost have of the header file and makes it difficult to look for polyline-specific methods.